x/mongo/driver/operation.go

```diff
@@ -871,14 +891,26 @@ func (Operation) decompressWireMessage(wm []byte) ([]byte, error) {
 }
 
 func (op Operation) createWireMessage(ctx context.Context, dst []byte,
-	desc description.SelectedServer, conn Connection) ([]byte, startedInformation, error) {
+	desc description.SelectedServer, srvr Server, conn Connection) ([]byte, startedInformation, error) {
+	// Calculate value of 'maxTimeMS' field to append to the wire message based on the current
+	// context's deadline and the 90th percentile RTT if op.Timeout is set.
+	var maxTimeMS *int64
+	if op.Timeout != nil {
+		if deadline, ok := ctx.Deadline(); ok {
+			remainingTimeout := time.Until(deadline)
+			maxTimeMSVal := int64(remainingTimeout/time.Millisecond) -
+				int64(srvr.RTT90()/time.Millisecond)
```

Comment:
A non-positive "remaining timeout" value used for `maxTimeMS` seems like an error condition since we're already at or past the deadline. We should return an error here in that case (maybe `ErrDeadlineWouldBeExceeded`).

Reply:
Ah good point. This logic has been moved to `Execute` and I've added a potential error return if the calculated `maxTimeMS` is < 0. It's still `ErrDeadlineWouldBeExceeded`, but the message is slightly different than the check directly before `roundTrip`.